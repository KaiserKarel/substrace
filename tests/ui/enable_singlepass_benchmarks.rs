#![allow(panics)]
// This should not be flagged
// #[cfg(feature = "runtime-benchmarks")]

// This should
#[cfg(feature = "runtime-benchmarks")]
fn a() {

}

// This should
#[cfg(any(feature = "runtime-benchmarks", other_key = "other-value"))]
fn b() {

}

// This should
#[cfg(any(yes = "yes", feature = "runtime-benchmarks"))]
fn c() {

}

// This should not
#[cfg(any(feature = "runtime-benchmarks", test))]
fn d() {

}

// This should not
#[cfg(any(test, feature = "runtime-benchmarks"))]
fn e() {

}

// This should
#[cfg(any(feature="runtime-benchmarks"))]
fn f() {

}

fn main() {

}