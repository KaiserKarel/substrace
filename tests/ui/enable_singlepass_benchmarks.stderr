error: substrace: benchmarks not run in tests.
       Found:
       #[cfg(feature = "runtime-benchmarks")]
       
       at line 6 in ./$DIR/enable_singlepass_benchmarks.rs. Suggested replacement:
       #[cfg(any(test, feature = "runtime-benchmarks"))]
   |
   = note: `-D enable-singlepass-benchmarks` implied by `-D warnings`

error: substrace: benchmarks not run in tests.
       Found:
       #[cfg(any(feature = "runtime-benchmarks", other_key = "other-value"))]
       
       at line 12 in ./$DIR/enable_singlepass_benchmarks.rs. Suggested replacement:
       #[cfg(any(test, feature = "runtime-benchmarks", other_key = "other-value"))]

error: substrace: benchmarks not run in tests.
       Found:
       #[cfg(any(yes = "yes", feature = "runtime-benchmarks"))]
       
       at line 18 in ./$DIR/enable_singlepass_benchmarks.rs. Suggested replacement:
       #[cfg(any(test, yes = "yes", feature = "runtime-benchmarks"))]

error: substrace: benchmarks not run in tests.
       Found:
       #[cfg(any(feature="runtime-benchmarks"))]
       
       at line 36 in ./$DIR/enable_singlepass_benchmarks.rs. Suggested replacement:
       #[cfg(any(test, feature="runtime-benchmarks"))]

error: aborting due to 4 previous errors

